packages/arb-rpc-node/snapshot/snapshot.go

```diff
@@ -372,42 +389,57 @@ func (s *Snapshot) GetStorageAt(account common.Address, index *big.Int) (*big.In
 }
 
 func (s *Snapshot) SetNonce(account common.Address, nonce uint64) error {
-	_, err := s.basicCall(arbos.SetNonceData(account, nonce), common.NewAddressFromEth(arbos.ARB_TEST_ADDRESS))
+	res, err := s.basicAddMessage(arbos.SetNonceData(account, nonce), common.NewAddressFromEth(arbos.ARB_TEST_ADDRESS))
 	if err != nil {
 		return err
 	}
+	if err := checkValidResult(res); err != nil {
+		return err
+	}
 	return nil
 }
 
 func (s *Snapshot) SetBalance(account common.Address, ballance *big.Int) error {
-	_, err := s.basicCall(arbos.SetBalanceData(account, ballance), common.NewAddressFromEth(arbos.ARB_TEST_ADDRESS))
+	res, err := s.basicAddMessage(arbos.SetBalanceData(account, ballance), common.NewAddressFromEth(arbos.ARB_TEST_ADDRESS))
 	if err != nil {
 		return err
 	}
+	if err := checkValidResult(res); err != nil {
+		return err
+	}
 	return nil
 }
```

Comment:
I think these new methods could be simpler by defining 
```golang
func (s *Snapshot) addArbTestMessage(data []byte) error {
	res, err := s.basicAddMessage(data), common.NewAddressFromEth(arbos.ARB_TEST_ADDRESS))
	if err != nil {
		return err
	}
	if err := checkValidResult(res); err != nil {
		return err
	}
	return nil
}
```
and then this just becomes
```go
func (s *Snapshot) SetBalance(account common.Address, balance *big.Int) error {
    return addArbTestMessage(arbos.SetBalanceData(account, balance))
}
```

Also, typo in `balance` 🙂